testing/e2e/metric_extraction_test.go

```diff
@@ -0,0 +1,56 @@
+package e2e
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/sensu/sensu-go/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMetricExtraction(t *testing.T) {
+	t.Parallel()
+
+	// Start the backend
+	backend, cleanup := newBackend(t)
+	defer cleanup()
+
+	// Initializes sensuctl
+	sensuctl, cleanup := newSensuCtl(backend.HTTPURL, "default", "default", "admin", "P@ssw0rd!")
+	defer cleanup()
+
+	// Start the agent
+	agentConfig := agentConfig{
+		ID:          "TestMetricExtraction",
+		BackendURLs: []string{backend.WSURL},
+	}
+	agent, cleanup := newAgent(agentConfig, sensuctl, t)
+	defer cleanup()
+
+	// Create a check that publish check requests
+	out, err := sensuctl.run(
+		"check", "create", "nagios-metric",
+		"--publish",
+		"--interval", "1",
+		"--subscriptions", "test",
+		"--command", "echo 'PING ok - Packet loss = 0% | percent_packet_loss=0'",
```

Comment:
This is the failure on windows only: `echo`

Reply:
We could create a go program in tools https://github.com/sensu/sensu-go/tree/master/tools that will allow us to write a string to stdout. That way, the check will be executed properly on both platforms

Reply:
Ah, good catch!